cedar-policy-validator/src/type_error.rs

```diff
@@ -331,9 +339,48 @@ pub(crate) enum UnexpectedTypeHelp {
 }
 
 /// Structure containing details about an incompatible type error.
-#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+#[derive(Diagnostic, Error, Debug, Clone, Hash, Eq, PartialEq)]
+#[diagnostic(help("{context} must have compatible types; {hint}"))]
```

Comment:
Nit. I think for other error messages we've preferred using a period, and then capitalizing the first letter of the next sentence (so this would require changes to `LubHelp` too).
```suggestion
#[diagnostic(help("{context} must have compatible types. {hint}"))]
```